helix-dap/src/client.rs

```diff
@@ -0,0 +1,647 @@
+use crate::{
+    transport::{Event, Payload, Request, Response, Transport},
+    Result,
+};
+use log::{error, info};
+use serde::{Deserialize, Serialize};
+use serde_json::{from_value, to_value, Value};
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+use std::{collections::HashMap, process::Stdio};
+use tokio::{
+    io::{AsyncBufRead, AsyncWrite, BufReader, BufWriter},
+    join,
+    net::TcpStream,
+    process::{Child, Command},
+    sync::{
+        mpsc::{channel, Receiver, Sender, UnboundedReceiver, UnboundedSender},
+        Mutex,
+    },
+};
+
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ColumnDescriptor {
```

Comment:
Do all these types need an extra file? I think so, will split out if you agree.

Reply:
I'm making some changes, will split all of these into a `types.rs`